core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorResponseFactory.java

```diff
@@ -10,10 +10,23 @@
 @Slf4j
 public class ErrorResponseFactory
 {
-    private final ConstraintViolationMapper constraintViolationMapper = new ConstraintViolationMapper();
-    private final RuleViolationMapper ruleViolationMapper = new RuleViolationMapper();
-    private final ClientErrorMapper clientErrorMapper = new ClientErrorMapper();
-    private final GenericErrorMapper genericErrorMapper = new GenericErrorMapper();
+    private final ConstraintViolationMapper constraintViolationMapper;
+    private final RuleViolationMapper ruleViolationMapper;
+    private final ClientErrorMapper clientErrorMapper;
+    private final GenericErrorMapper genericErrorMapper;
+
+    public ErrorResponseFactory()
```

Comment:
I was a bit confused. At first I thought this change would force any user of mizool to now implement the `GlobalParametersSupplier` so that dependecy injection does not derp out. Then I realized that none of the classes affected by this PR partake in dependency injection. Now I began to wonder how one would even make use of this new feature, and I did not find that code anywhere.
Long story short, the only way to make use of this feature is by using neither the `AbstractErrorHandlingFilter`, nor the `RestExceptionMapper` from mizool, but roll your own mechanism. Using these two base features will in fact prevent you from making use of this new feature.

This smells. It should be possible - but not mandatory - to make use of this feature in the servlet and/or JAX-RS world, too.

Reply:
Yeah, that's true. I'm also not really sure how to tackle this as all three known usages of the `ErrorResponseFactory`, the `AbstractErrorHandlingFilter`, the `RestExceptionMapper` and the `SystemDataResponseEntityExceptionHandler` rely on `new` rather than any form of injection.

As far as I can see, the `AbstractErrorHandlingFilter` and the `RestExceptionMapper` could have used injection. I don't know why they didn't. For the `SystemDataResponseEntityExceptionHandler`, should be possible as well, as long as I register the `ErrorResponseFactory` as a `@Bean` in some `@Configuration` class.

The question would be the "possible but not mandatory" part. If we want to solve this with injection, we need to have an injection point, which in turn means, we need to provide at least a noop-implementation that can be specialized inside the project. If I remember correctly, JavaEE allows this with the `@Specializes` annotation. I would have to investigate whether this also works for Spring.

Also, due to this topic, I'm beginning to wonder whether it was a good idea to use mizool for a Spring project in the first place or whether we should at least have forked this into "mizool-spring" or something.... 🤔

Reply:
The goal should actually be to have a core that is completely agnostic, and does nothing by itself. It can be used in SE, in EE, in Spring, in Microprofile, whatever. Because it is just plain java.

And then you have technologies for the specific environments, that wrap the generic logic contained inside the core into the specifica needed in this environment.

Reply:
Good point. Seeing that among others, there are the following dependencies (among others) in `mizool-core`:
```xml
<dependency>
    <groupId>javax.enterprise</groupId>
    <artifactId>cdi-api</artifactId>
    <version>1.2</version>
    <scope>provided</scope>
</dependency>
<dependency>
    <groupId>javax.servlet</groupId>
    <artifactId>javax.servlet-api</artifactId>
    <version>3.1.0</version>
    <scope>provided</scope>
</dependency>
<dependency>
    <groupId>javax.ws.rs</groupId>
    <artifactId>javax.ws.rs-api</artifactId>
    <version>2.0.1</version>
</dependency>
```
I would say, we messed that goal up real good.

But that being said, I think the `ErrorResponseFactory` is not that far off as it _is_ plan Java. This leaves the question on how we want to change the `AbstractErrorHandlingFilter` and the `RestExceptionMapper`.
The `JacksonErrorHandlingFilter`, that extends the `AbstractErrorHandlingFilter`, already uses field injection, so I could try to change the `AbstractErrorHandlingFilter` to use field injection as well.
If I remember correctly, our resources in majestix also use field injection, so we might be lucky that the `RestExceptionMapper` can also use field injection.

We would have to pair that approach with a `NoopGlobalParametersSupplier` that can be specialized by the application.

Reply:
Since the `RestExceptionMapper` resides in `mizool-core`, I'll have to provide said supplier in the core as well.

Reply:
I now have made the changes locally, but I don't want to push them without first figuring out how to test them in a JavaEE environment. I'm currently not sure how to launch either majestix or any other of our EE environment. I'll have to figure that out tomorrow.